opamp-client/src/main/java/io/opentelemetry/opamp/client/internal/OpampClient.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.opamp.client.internal;
+
+import io.opentelemetry.opamp.client.internal.response.MessageData;
+import opamp.proto.Opamp;
+
+public interface OpampClient {
```

Comment:
I don't see some of opamp-go Client's methods here, e.g. `SetAgentDescription`. Just wanted to confirm this is deliberate and not accidentally missing.

Reply:
Yes, for context, I'm currently moving the PoC implementation created [here](https://github.com/elastic/opamp-java/tree/main/client) which had only the bare minimum apis/functionality for our use-case to work. So we should keep on adding more methods in the future. The `SetAgentDescription` one seems to be pretty straightforward so I think we could add it right away to account for the use-case of changing the description after the client has started.

An interesting note on the client initialization method used in the PoC is that it relies on the builder pattern. So, setting the agent description before starting the client is done in the client builder, as you can see [here](https://github.com/elastic/opamp-java/blob/main/client/src/main/java/co/elastic/opamp/client/HttpOpampClientBuilder.java#L182). The are 2 main reasons for this, the first one being that it's less error-prone as users wouldn't need to remember setting the agent description before starting the client, which is something I noticed is mentioned [here](https://github.com/open-telemetry/opamp-go/blob/ad5317009abb490ff5e57e564ac8e82f70f9f477/client/client.go#L18) in the Go impl, and the second reason is that using the builder pattern to solve these issues is a common practice in Java and makes it easier to adjust/extend initialization-related processes in the future. It's also the pattern used when initializing the OTel Java SDK so I think that's a bonus in terms of consistency across Java-related projects in OTel.

Reply:
I've just added `SetAgentDescription`.

Reply:
Makes sense. I think you are free to deviate from opamp-go and use whatever is the most idiomatic approach for Java. Builder pattern is less common in Go world. Struct of options is typically what Go uses, e.g. the standard library http server: https://pkg.go.dev/net/http#hdr-Servers